### ISSUE TITLE
TrustManagerFactory PKIX implementation not found

### ISSUE THREAD

**Issue by rocket-pig** [vector-im/riot-android#840]
HTC Ruby "Amaze 4G" with cyanogenmod 10 official "cm-10-20131006-NIGHTLY-ruby"

from a fresh install. only thing I did was install f-droid and use it to install riot. I can try on the dev build but my guess is it will be same.

During install I saw this go by and seemed possibly related. I am not incredibly knowlegable so could be wrong.

D/PackageManager(  401): New package installed in /data/app/im.vector.alpha-1.apk
W/PackageManager(  401): Not granting permission android.permission.READ_LOGS to package im.vector.alpha (protectionLevel=50 $
W/PackageManager(  401): Unknown permission android.permission.MANAGE_DOCUMENTS in package im.vector.alpha
W/PackageManager(  401): Unknown permission com.sec.android.provider.badge.permission.READ in package im.vector.alpha
W/PackageManager(  401): Unknown permission com.sec.android.provider.badge.permission.WRITE in package im.vector.alpha
W/PackageManager(  401): Unknown permission com.htc.launcher.permission.READ_SETTINGS in package im.vector.alpha
W/PackageManager(  401): Unknown permission com.htc.launcher.permission.UPDATE_SHORTCUT in package im.vector.alpha
W/PackageManager(  401): Unknown permission com.sonyericsson.home.permission.BROADCAST_BADGE in package im.vector.alpha
W/PackageManager(  401): Unknown permission com.anddoes.launcher.permission.UPDATE_COUNT in package im.vector.alpha
W/PackageManager(  401): Unknown permission com.majeur.launcher.permission.UPDATE_BADGE in package im.vector.alpha
I/RegisteredComponentCache(  690): ComponentInfo: ResolveInfo{40f286f0 com.android.apps.tag.TagViewer p=0 o=0 m=0x108000}, te$

I have two ongoing encrypted 'rooms' with both another android client and an iOS one. There are no errors concerning chat encryption. Only when trying to view attached photos. It goes a little something like this:

I/ActivityManager(  401): START {cmp=im.vector.alpha/im.vector.activity.VectorMediasViewerActivity (has extras) u=0} from pid 8390
D/VectorApp( 8390): onActivityPaused im.vector.activity.VectorRoomActivity@41b0b000
D/VectorApp( 8390): ## setCurrentActivity() : from im.vector.activity.VectorRoomActivity@41b0b000 to null
D/VectorApp( 8390): ## startActivityTransitionTimer()
D/VectorApp( 8390): onActivityCreated im.vector.activity.VectorMediasViewerActivity@41f47068
D/VectorApp( 8390): onActivityStarted im.vector.activity.VectorMediasViewerActivity@41f47068
D/VectorApp( 8390): onActivityResumed im.vector.activity.VectorMediasViewerActivity@41f47068
D/VectorApp( 8390): ## setCurrentActivity() : from null to im.vector.activity.VectorMediasViewerActivity@41f47068
D/VectorApp( 8390): ## stopActivityTransitionTimer()
W/ResourceType(  401): Skipping entry 0x7f04000a in package table 0 because it is not complex!
I/ActivityManager(  401): Displayed im.vector.alpha/im.vector.activity.VectorMediasViewerActivity: +181ms
D/MXMediaDwndWorkerTk( 8390): bitmapForURL() : file_8b43f68025d28332e814699c7b686eb16e0f0b8d does not exist
D/MXMediaDwndWorkerTk( 8390): MXMediaDownloadWorkerTask org.matrix.androidsdk.db.MXMediaDownloadWorkerTask@41cd3b98 starts
E/CertUtil( 8390): ## newPinnedSSLSocketFactory() : TrustManagerFactory.getInstance failed TrustManagerFactory PKIX implementation not found
D/MP-Decision(  942): UP Nw:2.700000 Tw:180 rq:4.400000 seq:199
D/VectorApp( 8390): onActivitySaveInstanceState im.vector.activity.VectorRoomActivity@41b0b000
D/VectorApp( 8390): onActivityStopped im.vector.activity.VectorRoomActivity@41b0b000
D/MP-Decision(  942): DOWN Ns:2.100000 Ts:270 rq:0.000000 seq:294 (Up time:294)
D/MXMediaDwndWorkerTk( 8390): publishProgress org.matrix.androidsdk.db.MXMediaDownloadWorkerTask@41cd3b98 : 1
D/MXMediaDwndWorkerTk( 8390): publishProgress org.matrix.androidsdk.db.MXMediaDownloadWorkerTask@41cd3b98 : 1
D/MXMediaDwndWorkerTk( 8390): publishProgress org.matrix.androidsdk.db.MXMediaDownloadWorkerTask@41cd3b98 : 1
D/MXMediaDwndWorkerTk( 8390): publishProgress org.matrix.androidsdk.db.MXMediaDownloadWorkerTask@41cd3b98 : 1

....deleted thousands of repeats publishProgress....

D/MXMediaDwndWorkerTk( 8390): publishProgress org.matrix.androidsdk.db.MXMediaDownloadWorkerTask@41cd3b98 : 97
D/MXMediaDwndWorkerTk( 8390): publishProgress org.matrix.androidsdk.db.MXMediaDownloadWorkerTask@41cd3b98 : 97
D/MXMediaDwndWorkerTk( 8390): publishProgress org.matrix.androidsdk.db.MXMediaDownloadWorkerTask@41cd3b98 : 99
D/dalvikvm( 8390): GC_FOR_ALLOC freed 2863K, 43% free 16922K/29187K, paused 69ms, total 69ms
I/dalvikvm-heap( 8390): Grow heap (frag case) to 28.361MB for 2031632-byte allocation
E/MXEncryptAtt( 8390): ## decryptAttachment() :  Digest value mismatch
D/MXMediaDwndWorkerTk( 8390): The download org.matrix.androidsdk.db.MXMediaDownloadWorkerTask@41cd3b98failed.

**Comment by rocket-pig** [comment]
Meanwhile on iOS device I will frequently get "Unknown inbound session ID" from the one client that is not on the LAN. The encryption information popup will say that the not-LAN user has no Curve ID key, no ED25519 fingerprint, "unknown device". But the local android device will still be able to receive and send encrypted text.

**Comment by rocket-pig** [comment]
I think mine is a duplicate of https://github.com/vector-im/riot-android/issues/812

**Comment by rocket-pig** [comment]
I found another rom with android K "4.4.4" and picture loading works. From the gallons of reading I did, it appears secure key storage was going thru a lot of changes right around 4.1 to 4.3. I'm not an expert, all I know is this fixed the issue.

relevant stuff from log:

W/AwContents( 1813): nativeOnDraw failed; clearing to background color.
W/ResourceType(  543): Skipping entry 0x7f04000a in package table 0 because it is not complex!
D/MXMediaDwndWorkerTk( 1813): bitmapForURL() : file_88f29d3abcd2534ac2bdda05d1afa635e66141dd does not exist
D/MXMediaDwndWorkerTk( 1813): MXMediaDownloadWorkerTask org.matrix.androidsdk.db.MXMediaDownloadWorkerTask@420e7ac0 starts
I/ActivityManager(  543): Displayed im.vector.alpha/im.vector.activity.VectorMediasViewerActivity: +337ms
I/Timeline( 1813): Timeline: Activity_idle id: android.os.BinderProxy@41b08710 time:851051
I/Timeline(  543): Timeline: Activity_windows_visible id: ActivityRecord{419b7e00 u0 im.vector.alpha/im.vector.activity.VectorMediasViewerActivity t3} time:851255
D/VectorApp( 1813): onActivitySaveInstanceState im.vector.activity.VectorRoomActivity@41963980
D/VectorApp( 1813): onActivityStopped im.vector.activity.VectorRoomActivity@41963980
W/AwContents( 1813): nativeOnDraw failed; clearing to background color.
W/AwContents( 1813): nativeOnDraw failed; clearing to background color.
W/AwContents( 1813): nativeOnDraw failed; clearing to background color.
W/AwContents( 1813): nativeOnDraw failed; clearing to background color.
W/AwContents( 1813): nativeOnDraw failed; clearing to background color.
D/MXEncryptAtt( 1813): Decrypt in 3 ms
D/MXMediaDwndWorkerTk( 1813): The download org.matrix.androidsdk.db.MXMediaDownloadWorkerTask@420e7ac0is done.
